### PR TITLE
Send & Get Datasets

### PR DESCRIPTION
**Pull request by abogaziah**
first milestone in the issue #2670

### CODE REVIEW

File: syft/frameworks/torch/fl/dataset.py [hunk 1]
```diff
@@ -22,8 +24,19 @@ class BaseDataset:
...
+    def __init__(
+        self,
+        data,
+        targets,
+        transform=None,
+        id: int = None,
+        owner=None,
+        tags: List[str] = None,
+        description: str = None,
+        child=None,
```
Comment (LaRiffle) [inline syft/frameworks/torch/fl/dataset.py]: for conciseness just put `**kwargs` and report them in `super().__init__(**kwargs)`

Comment (abogaziah) [reply syft/frameworks/torch/fl/dataset.py]: done

File: syft/generic/pointers/pointer_dataset.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            id_at_location=id_at_location,
+            owner=owner,
+            garbage_collect_data=garbage_collect_data,
+            id=id,
+            tags=tags,
+            description=description,
+        )
+
+    def get(self):
+        dataset = self.location.get_obj(self.id_at_location)
```
[the remark this answers is not in the document]
Comment (abogaziah) [reply syft/generic/pointers/pointer_dataset.py]: Ok, I changed it to use the inherited .get() functionality of ObjectPointer

File: syft/workers/base.py [hunk 3]
```diff
@@ -951,7 +951,7 @@ def search(self, query: Union[List[Union[str, int]], str, int]) -> List[PointerT
                 # set garbage_collect_data to False because if we're searching
                 # for a tensor we don't own, then it's probably someone else's
                 # decision to decide when to delete the tensor.
-                ptr = obj.create_pointer(garbage_collect_data=False, owner=sy.local_worker).wrap()
+                ptr = obj.create_pointer(garbage_collect_data=False, owner=sy.local_worker, location=self).wrap()
```
Comment (LaRiffle) [inline syft/workers/base.py]: Why did you need to change this `location` argument?

Comment (abogaziah) [reply syft/workers/base.py]: because if the location wasn't passed it would assume the location is the local worker, which isn't correct. the search result would be always me | id -> me | id_at_location
